lib/aws/rails/sqs_active_job/executor.rb

```diff
@@ -36,8 +42,31 @@ def execute(message)
               job_msg = job ? "#{job.id}[#{job.class_name}]" : 'unknown job'
               @logger.info "Error processing job #{job_msg}: #{e}"
               @logger.debug e.backtrace.join("\n")
+            ensure
+              refresh_monitor = false
             end
           end
+
+          @monitor.post(message) do |message|
+            while refresh_monitor
+              begin
+                # Extend the visibility timeout to the provided refresh timeout
+                message.change_visibility({ visibility_timeout: @visibility_refresh })
```

Comment:
If this is lower, what is submitted to change_visibility will be respected first.

Reply:
I would expect this to throw an error since 12 hours is the max. We can't extend past that.

Additionally, I'm thinking of this scenario:

1. `visiblity_timeout` is 2 hours, `visbility_refresh` is 1 hour
2. we receive a message, and start processing. `visibility_timeout` is extended to 3 hours
3. sleep for 30 minutes
4. 30 minutes elapses (30 mins total), `visibility_timeout` is extended to 4 hours
5. 30 minutes elapses (1 hour total), `visibility_timeout` is extended to 5 hours
6. ...

In the above scenario, won't we hit the limit much faster than the time the job has to run? Basically what I'm trying to say is that we should cover the case where the `visiblity_timeout` has already maxed out.